### ISSUE TITLE
Need help troubleshooting LDAP install

### ISSUE THREAD

**Issue by mikezilber** [ibm-cloud-architecture/cloudpak8s#94]
I am trying to execute these directions in Fyre.
Frankly, I don't know ANYTHING about Directory server on linux. I've installed it on windows once a long time ago.

I am hitting errors in this step:
[root@odcicpforautomation-inf bin]# ./ldapsearch -h ldap://odcicpforautomation-inf.fyre.ibm.com:389 -s base -b " " objectclass=* ibm-slapdisconfigurationmode
ldap_simple_bind: Can't contact LDAP server
[root@odcicpforautomation-inf bin]#

can someone help me troubleshoot the issue?

**Comment by mikezilber** [comment]
When I start the server, I get this:

[root@odcicpforautomation-inf sbin]# ./ibmslapd -I dsinst1
GLPSRV041I Server starting.
GLPSRV235I Premium feature activation code is loaded and initialized.
GLPSRV012E A primary administrator DN and administrator PW must be specified in the server configuration.

I got it fixed by running:
./idsdnpw -u cn=<usrer> -p <password>

then running 
 ./ibmslapd -I dsinst1

this time it went OK

**Comment by dthiesse** [comment]
@mikezilber It sounds like you have figured out the answer here. Is that correct? If so, I'm happy. Can I close your issue?

**Comment by mikezilber** [comment]
@dthiesse Not exactly.  I got the LDAP up, but I am not able to import the LDIF file provided.
There are bunch of commands missing/invalid in the doc.  I am slowing working through the issues and will update it.  BUT if you can help, it would save me some time.

**Comment by dthiesse** [comment]
Sorry to hear @mikezilber . I am assigning your issue to @pberlandier who leads the CP for Automation team to see if he can help.

**Comment by mikezilber** [comment]
@dthiesse thanks!
@pberlandier, I hit several issues while installing IDS.
1.  the ldap instance name had to be the same as DB2 instance name.  Not sure why.
2.  bunch or users had to share the same groups.  Have to look at my notes.
3.  Once I have the LDAP installed, I still could not import the LDIF file linked in the cookbook,
Even adding "o=ibm,c=US" as suffix, I still was not able to import it.  Error, somethink like "c=UC" entry not found.  
I was able to follows these direction.
https://www.ibm.com/developerworks/aix/library/au-ssl_config/index.html
Note: this is the first time I am dealing with installing IDS on linux.

**Comment by mikezilber** [comment]
I guess, if IDS can work with out DB2 connection. The issue is importing LDIF file.

**Comment by pberlandier** [comment]
@mikezilber So, you're running through this page, right? https://ocp42.cloudpak8s.io/automation/shared-services/

The ldif file is just an example to validate the directory is working as expected. Did you try another ldif file?

**Comment by mikezilber** [comment]
well, I tried the 1st one from the link above.  After I created a suffix it worked.

**Comment by dthiesse** [comment]
Hi @mikezilber @pberlandier . Since this issue has been open for a month, I'd like to be able to either close it or get the necessary updates on the site. @mikezilber do you think we need to document to add a suffix as you describe worked for you? If so, and you lack the time or experience to create your own PR following the contribution guide [here](https://ocp42.cloudpak8s.io/contribute/), please let me know the exact wording and I will do it. Thanks for your help.

**Comment by mikezilber** [comment]
So, I did not have a chance to tear down and retry.  adding "o=ibm" suffix and modifying the the ldif file to remove c=US references worked for me.

**Comment by mikezilber** [comment]
frankly, it is very RARE that we are asked to set up LDAP.  For POC/DEMO purposes we should document OpenLDAP since it can run in containers.
